### ISSUE TITLE
[Bug]: ConnectivityResult.vpn missing

### ISSUE THREAD

**Issue by imdatsolak** [fluttercommunity/plus_plugins#1259]
### Platform

macOS

Project is for iOS, Android, macOS, Windows, Linux

### Plugin

connectivity_plus

### Version

3.0.0

### Flutter SDK

3.3.4

### Steps to reproduce

1. Add package to own project
2. Try building macOS Version
3. Error-Message during compile:

``` bash
../../../../.pub-cache/hosted/pub.dartlang.org/connectivity_plus-3.0.0/lib/src/connectivity_plus_linux.dart:54:33: Error: Member not found: 'vpn'.
      return ConnectivityResult.vpn;
```

### Code Sample

```dart
None required
```

### Logs

```shell
Compile-Error
```

### Flutter Doctor

```shell
[✓] Flutter (Channel stable, 3.3.4, on macOS 12.6 21G115 darwin-arm, locale en-US)
    • Flutter version 3.3.4 on channel stable at /Users/iso/Development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision eb6d86ee27 (13 days ago), 2022-10-04 22:31:45 -0700
    • Engine revision c08d7d5efc
    • Dart version 2.18.2
    • DevTools version 2.15.0

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/iso/Library/Android/sdk
    • Platform android-33, build-tools 30.0.3
    • Java binary at: /Users/iso/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/213.7172.25.2113.9014738/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14A400
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] Android Studio (version 2021.3)
    • Android Studio at /Users/iso/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/213.7172.25.2113.9014738/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] VS Code (version 1.71.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.48.0

[✓] Connected device (7 available)
    • SM T860 (mobile)   • R52N8175P2E                              • android-arm64  • Android 12 (API 31)
    • SM A536B (mobile)  • RZCT909TEZY                              • android-arm64  • Android 12 (API 31)
    • MiniMe (mobile)    • 00008020-000108D13A41002E                • ios            • iOS 15.7 19H12
    • iPhone X (mobile)  • dd8204853d6bcfce9efecee2fc094c65327e0b1e • ios            • iOS 16.0.3 20A392
    • iPhone 6s (mobile) • ab7504427aab66aece9687552a8a6fc2665f980a • ios            • iOS 15.7 19H12
    • macOS (desktop)    • macos                                    • darwin-arm64   • macOS 12.6 21G115 darwin-arm
    • Chrome (web)       • chrome                                   • web-javascript • Google Chrome 105.0.5195.102
    ! Error: MiniMe is busy: Fetching debug symbols for MiniMe. Xcode will continue when MiniMe is finished. (code -10)
    ! Error: iPhone 6s is busy: Fetching debug symbols for iPhone 6s. Xcode will continue when iPhone 6s is finished. (code -10)

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

### Checklist before submitting a bug

- [X] I Google'd a solution and I couldn't find it
- [X] I searched on StackOverflow for a solution and I couldn't find it
- [X] I read the README.md file of the plugin
- [X] I'm using the latest version of the plugin
- [X] All dependencies are up to date with `flutter pub upgrade`
- [X] I did a `flutter clean`
- [X] I tried running the example project

**Comment by husainazkas** [comment]
Have same issue while compile into android app bundle for releasing. Please ensure that enum value is valid because there is no ```ConnectivityResult.vpn``` found. Thanks a lot for every contributors.

**Comment by imdatsolak** [comment]
I reverted back to the previous version until this is fixed. Thank you a million for all contributors!

**Comment by zeewell** [comment]
It may be necessary to wait for a new version of connectivity_plus_platform_interface to be released.

@vbuberen When will the new version of connectivity_plus_platform_interface be released? Looking forward to it!

**Comment by vbuberen** [comment]
> @vbuberen When will the new version of connectivity_plus_platform_interface be released? Looking forward to it!

There were no new features or critical fixes introduced in the 3.0.0 release. Use older version of the plugin and don't tag people multiple times. Release will be when maintainers have time. It is not a commercial project, so nobody is going to promise any deadlines, especially when nobody blocks you from using older version of the plugin.

**Comment by zeewell** [comment]
I am very sorry, my words may not be polite, but it was not my intention.

> > @vbuberen When will the new version of connectivity_plus_platform_interface be released? Looking forward to it!
> 
> There were no new features or critical fixes introduced in the 3.0.0 release. Use older version of the plugin and don't tag people multiple times. Release will be when maintainers have time. It is not a commercial project, so nobody is going to promise any deadlines, especially when nobody blocks you from using older version of the plugin.

**Comment by mariopepe** [comment]
No worries at all @zeewell  you have been very polite!
I think is vbuberen that is having a stressful day!

**Comment by miquelbeltran** [comment]
This is weird, because the `vpn` enum type exists in the file `src/enums.dart` of the `connectivity_plus_platform_interface`:

https://github.com/fluttercommunity/plus_plugins/blob/main/packages/connectivity_plus/connectivity_plus_platform_interface/lib/src/enums.dart#L19

And the internal package 1.2.1 has been published: https://pub.dev/packages/connectivity_plus_platform_interface

And also, the project does not complain when opened in the IDE.

However, if I do the following, I can reproduce the issue:

1. Create a new project with `flutter create error_1259`
2. cd `error_1259`
3. `flutter pub add connectivity_plus`
4. `flutter run`

I get the same error as the original poster.

**Comment by miquelbeltran** [comment]
Forcing the platform interface package to be "from git" fixes the issue. So I guess there was a bogus release. I will force the publishing of a new version of the platform interface on pub.

```
dependency_overrides:
  connectivity_plus_platform_interface:
    git:
      url: https://github.com/fluttercommunity/plus_plugins.git
      path: packages/connectivity_plus/connectivity_plus_platform_interface
```

**Comment by miquelbeltran** [comment]
I published `connectivity_plus_platform_interface` 1.2.2 and the issue should be fixed now.

**Comment by imdatsolak** [comment]
Thanks a million.
